clang/include/clang/APINotes/APINotesReader.h

```diff
@@ -174,14 +174,16 @@ class APINotesReader {
   /// \param name The name of the global variable.
   ///
   /// \returns information about the global variable, if known.
-  VersionedInfo<GlobalVariableInfo> lookupGlobalVariable(llvm::StringRef name);
+  VersionedInfo<GlobalVariableInfo>
+  lookupGlobalVariable(std::optional<Context> context, llvm::StringRef name);
```

Comment:
I think that swapping the order of the parameters would be better.  If you expect `context` to always be provided, then why have `std::optional<Context>`?  Otherwise, if it is optional, why not make the signature:

```
VersionedInfo<GlobalVariableInfo>
lookupGlobalVariable(llvm::StringRef name, std::optional<Context> context = none);
```

Similar for the other changes.

Reply:
Thanks, changed the signatures as you suggested.

> If you expect context to always be provided, then why have std::optional<Context>?

It won't be provided for symbols in the global context.